src/inMemory.ts

```diff
@@ -42,14 +46,22 @@ export class InMemoryTransport implements Transport {
   }
 
   async send(message: JSONRPCMessage): Promise<void> {
+    return this.sendWithAuth(message);
+  }
+
+  /**
+   * Sends a message with optional auth info.
+   * This is useful for testing authentication scenarios.
+   */
+  async sendWithAuth(message: JSONRPCMessage, authInfo?: AuthInfo): Promise<void> {
```

Comment:
This can be added directly to `send`, and I'd wrap `authInfo` in an object to make it a named parameter:
```suggestion
  async send(message: JSONRPCMessage, options?: { authInfo?: AuthInfo }): Promise<void>
```

This definition is still compatible with `Transport.send`.